lib/Sema/CSDiag.cpp

```diff
@@ -7624,6 +7624,18 @@ void FailureDiagnosis::diagnoseAmbiguity(Expr *E) {
       return;
   }
 
+  // Diagnose use of AnyObject as root for a keypath
+  if (auto KPE = dyn_cast<KeyPathExpr>(E)) {
```

Comment:
Please make this a constraint fix, let's not add new code to the expression diagnostics.

Reply:
@xedin Good idea! Done :)